packages/fleet/lib/models/nodes.integration.test.ts

```diff
@@ -0,0 +1,151 @@
+import { expect, describe, it, beforeEach, afterEach } from 'vitest';
+import * as nodes from './nodes.js';
+import * as deployments from './deployments.js';
+import { nodeStates } from '../types.js';
+import type { NodeState, Node, RoutingId } from '../types.js';
+import { getTestDbClient } from '../db/helpers.test.js';
+import type { knex } from 'knex';
+import { nanoid } from '@nangohq/utils';
+import { generateCommitHash } from './helpers.test.js';
+
+const activeDeployment = generateCommitHash();
+
+describe('Nodes', () => {
+    const dbClient = getTestDbClient('nodes');
+    const db = dbClient.db;
+    beforeEach(async () => {
+        await dbClient.migrate();
+        await deployments.create(db, activeDeployment);
+    });
+
+    afterEach(async () => {
+        await dbClient.clearDatabase();
+    });
+
+    it('should be successfully created', async () => {
+        const node = (
+            await nodes.create(db, {
+                routingId: 'my-routing-id',
+                deploymentId: activeDeployment,
+                url: 'http://localhost:3000',
+                image: 'nangohq/my-image:latest',
+                cpuMilli: 500,
+                memoryMb: 1024,
+                storageMb: 512
+            })
+        ).unwrap();
+        expect(node).toMatchObject({
```

Comment:
nb: just in case you don't know, toMatchObject is partial matching, usually I prefer to use strictEqual so there is no missing fields